### PR TITLE
Adding a global setting to skip L2 votes

### PR DESCRIPTION
**Pull request by brickpop**

### CODE REVIEW

**Comment by jordaniza** [comment]
Qu: would you add skipL2 to the event here?

```js
    function _updateOptimisticGovernanceSettings(OptimisticGovernanceSettings calldata _governanceSettings) internal {
        // Require the minimum veto ratio value to be in the interval [0, 10^6], because `>=` comparision is used.
        if (_governanceSettings.minVetoRatio == 0) {
            revert RatioOutOfBounds({limit: 1, actual: _governanceSettings.minVetoRatio});
        } else if (_governanceSettings.minVetoRatio > RATIO_BASE) {
            revert RatioOutOfBounds({limit: RATIO_BASE, actual: _governanceSettings.minVetoRatio});
        }

        // MinDuration is not constrained on the lower side, since the emergency plugin needs to submit
        // direct proposals after a super majority has approved them.

        // StandardProposalCondition.sol ensures that all proposals created via the standard Multisig have
        // the expected minimum duration.

        if (_governanceSettings.minDuration > 365 days) {
            revert MinDurationOutOfBounds({limit: 365 days, actual: _governanceSettings.minDuration});
        }

        governanceSettings = _governanceSettings;

        emit OptimisticGovernanceSettingsUpdated({
            minVetoRatio: _governanceSettings.minVetoRatio,
            minDuration: _governanceSettings.minDuration,
            l2AggregationGracePeriod: _governanceSettings.l2AggregationGracePeriod,
            l2InactivityPeriod: _governanceSettings.l2InactivityPeriod
        });
    }
```

File: .env.example [hunk 1]
```diff
@@ -9,6 +9,7 @@ DEPLOY_AS_PRODUCTION=true  # With false, the script will deploy mock helpers
 MIN_VETO_RATIO="300000" # 30% 
 L2_INACTIVITY_PERIOD="600" # in seconds (10 minutes)
 L2_AGGREGATION_GRACE_PERIOD="172800" # in seconds (2 days)
+SKIP_L2=true
```
[the remark this answers is not in the document]
Comment (brickpop) [reply .env.example:12]: For the first version, this should probably be set to `true`,  which it is already. 

This is a plugin setting that will be set to `true` on the initial phase, but be disabled on the next one. The DAO may also override this setting at any time. 

Do you think that people may get confused by `SKIP_L2=true` on `.env.example`?

File: src/OptimisticTokenVotingPlugin.sol [hunk 2]
```diff
@@ -42,6 +42,7 @@ contract OptimisticTokenVotingPlugin is
         uint64 minDuration;
         uint64 l2InactivityPeriod;
         uint64 l2AggregationGracePeriod;
+        bool skipL2;
```
[the remark this answers is not in the document]
Comment (brickpop) [reply src/OptimisticTokenVotingPlugin.sol:45]: Good catch

File: test/helpers/DaoBuilder.sol [hunk 3]
```diff
@@ -42,6 +42,7 @@ contract DaoBuilder is Test {
     uint64 public minDuration = 4 days;
     uint64 public l2InactivityPeriod = 10 minutes;
     uint64 public l2AggregationGracePeriod = 2 days;
+    bool public skipL2 = false;
```
[the remark this answers is not in the document]
Comment (brickpop) [reply test/helpers/DaoBuilder.sol:45]: Good catch again.

Yes, this is intentional here. Following an "empty by default" approach with the builder pattern, no setting is set unless explicitly altered.

**Comment by brickpop** [comment]
> Qu: would you add skipL2 to the event here?

Good one. Yes, we most likely should.

File: .env.example (hunk 1 above)
[the remark this answers is not in the document]
Comment (jordaniza) [reply .env.example:12]: Probably zero harm in adding the comment

**Review by jordaniza — APPROVED** [review]

**Pull request merged**